src/view/configuration/configurationView.jsx

```diff
@@ -267,60 +269,76 @@ const InstancesSection = ({ initInfo, context }) => {
-                                      }}
-                                      autoFocus
-                                    >
-                                      Delete
-                                    </Button>
-                                  </ButtonGroup>
-                                </Dialog>
-                              )}
-                            </DialogTrigger>
+                            <Button
+                              data-test-id="deleteInstanceButton"
+                              icon={<DeleteIcon />}
+                              variant="secondary"
+                              disabled={instances.length === 1}
+                              onPress={() => {
+                                setInstanceToDelete(index);
+                                setDeleteDialogOpen(true);
+                              }}
+                            >
+                              Delete instance
+                            </Button>
                           </View>
                         )}
                       </Item>
                     );
                   })}
                 </TabPanels>
               </Tabs>
+              <DialogContainer
+                onDismiss={() => {
+                  setDeleteDialogOpen(false);
+                  setInstanceToDelete(null);
+                }}
+              >
+                {deleteDialogOpen && (
+                  <Dialog data-test-id="resourceUsageDialog">
+                    <HeadingSlot>Resource Usage</HeadingSlot>
+                    <Divider />
+                    <Content>
+                      <Text>
+                        Any rule components or data elements using this instance
+                        will no longer function as expected when running on your
+                        website. We recommend removing these resources or
+                        switching them to use a different instance before
+                        publishing your next library. Would you like to proceed?
+                      </Text>
+                    </Content>
+                    <ButtonGroup>
+                      <Button
+                        data-test-id="cancelDeleteInstanceButton"
+                        variant="secondary"
+                        onPress={() => {
+                          setDeleteDialogOpen(false);
+                          setInstanceToDelete(null);
+                        }}
+                      >
+                        Cancel
+                      </Button>
+                      <Button
+                        data-test-id="confirmDeleteInstanceButton"
+                        variant="cta"
+                        onPress={() => {
+                          arrayHelpers.remove(instanceToDelete);
+                          setSelectedTabKey(
+                            String(
+                              instanceToDelete > 0 ? instanceToDelete - 1 : 0,
```

Comment:
This solves the issue number 1: Previously we were selecting the tab with the same number as the one that was deleted. So if you deleted the last tab, we were selecting a tab that was not existing and see a blank page.

The new logic is: 
- when you delete the first tab, we select the new first tab (the 2nd before the delete action)
- when we delete any other tab, we select the next from the left.